pkg/finder/finder.go

```diff
@@ -0,0 +1,87 @@
+package finder
+
+import (
+	"fmt"
+	"io/fs"
+	"log/slog"
+	"os"
+	"path/filepath"
+	"slices"
+
+	"github.com/getsops/sops/v3"
+	"github.com/getsops/sops/v3/stores/dotenv"
+	"github.com/getsops/sops/v3/stores/ini"
+	"github.com/getsops/sops/v3/stores/json"
+	"github.com/getsops/sops/v3/stores/yaml"
+)
+
+var supportedExtensions = []string{".ini", ".env", ".yaml", ".yml", ".json"}
+
+func getStore(fileType string) (sops.Store, error) {
+	switch fileType {
+	case ".ini":
+		return &ini.Store{}, nil
+	case ".env":
+		return &dotenv.Store{}, nil
+	case ".yaml", ".yml":
+		return &yaml.Store{}, nil
+	case ".json":
+		return &json.Store{}, nil
+	default:
+		return nil, fmt.Errorf("Unsupported file type: %s", fileType)
+	}
+}
+
+// SopsFiles searches a directory for YAML files and checks if they are valid SOPS files.
+// It returns a map of filepath with its sops.Metadata as value and an error.
+func SopsFiles(root string) (map[string]sops.Metadata, error) {
```

Comment:
Since maps have a non-deterministic order, how about returning a `[]SopsFile`, where `SopsFile` is defined as:

```go
type SopsFile struct {
  Path     string
  Metadata sops.Metadata
}
```

That way, we would avoid having to convert the map into a slice and sort it later on in order to have deterministic ordering when processing the files and printing the results to the user).

Reply:
I would also suggest a different name for the function. Always think how this would read when used from a different package. I find `finder.SopsFiles` a bit awkward. Maybe `finder.FindSopsFiles` could be a better alternative.

Another alternative could be to name the package `sops` and the function just `FindFiles`.